R/content-type.R

```diff
@@ -0,0 +1,115 @@
+#' Check the content type of a response
+#'
+#' A different content type than expected often leads to an error in parsing
+#' the response body. This function checks that the content type of the response
+#' is as expected and fails otherwise.
+#'
+#' @param resp A response object.
+#' @param valid_types A character vector of valid content types.
+#' @param valid_suffix A string given an "structured media type" suffix.
```

Comment:
This keeps the type and suffixes orthogonal in the way I think matches the spec.